### PR TITLE
[WIP][NEW] Record and send audio messages

### LINKED ISSUE #735: [NEW] Allow user to record an audio message and send it
We need a way to record and send audio messages without leaving the app.

Like so:
media

Instead of a swipe to cancel like in WhatsApp, for now we can have a UIAlertView with Send and Cancel options after the user raises his finger from the mic button.

Where to look:
[ChatControllerUploader.swift](https://github.com/RocketChat/Rocket.Chat.iOS/blob/master/Rocket.Chat/Controllers/Chat/ChatControllerUploader.swift)
[ChatViewController.swift](https://github.com/RocketChat/Rocket.Chat.iOS/blob/master/Rocket.Chat/Controllers/Chat/ChatViewController.swift)

### PR DESCRIPTION
**Pull request by falcaoaugustos**
@RocketChat/ios

Closes #735

I implemented the audio message recorder object and the toolbar right button selectors as text message sender and audio message recorder.

TODO: 
- [x] Localized Strings.

### CODE REVIEW

**Comment by falcaoaugustos** [comment]
I thought the project didn't need the SlackTextViewController pod new version because I had had a problem with CocoaPods.

I have done a new test and I noticed that the new version of the SlackTextViewController pod is required to the project work properly.

You can find the new version of the SlackTextViewController pod in:
repo: ```https://github.com/falcaoaugustos/SlackTextViewController.git ```
branch: ```feature/active_toolbar_right_button```

I am sending a PR to ```https://github.com/rafaelks/SlackTextViewController.git``` too.

Thank you.

**Comment by falcaoaugustos** [comment]
I implemented these improvements:

- [x] 1. Info.plist key
- [x] 2. Ask permission only when user tap into the record button
- [x] 3. Background circle view feedback
- [x] 3. Vibration feedback
- [x] Audio message timer

Thanks for your feedback.

File: Podfile [hunk 1]
```diff
@@ -15,7 +15,7 @@ def shared_pods
 
   # UI
   pod 'SideMenuController', :git => 'https://github.com/rafaelks/SideMenuController.git'
-  pod 'SlackTextViewController', :git => 'https://github.com/rafaelks/SlackTextViewController.git', :branch => 'chore/swift4_xcode9_ios11'
+  pod 'SlackTextViewController', :git => 'https://github.com/falcaoaugustos/SlackTextViewController.git', :branch => 'feature/active_toolbar_right_button'
```
Comment (rafaelks) [inline Podfile]: Ups...

File: Rocket.Chat/Info.plist [hunk 2]
```diff
@@ -72,6 +72,8 @@
 	<string>To take photos and upload it</string>
 	<key>NSPhotoLibraryUsageDescription</key>
 	<string>To upload photos from your library</string>
+	<key>NSMicrophoneUsageDescription</key>
+	<string>To record audio messages and upload it</string>
```
Comment (rafaelks) [inline Rocket.Chat/Info.plist]: 👍

**Comment by falcaoaugustos** [comment]
Hi @cardoso, I updated:

- [x] Audio record alert message

I sent a PR  on Rafael fork of the SlackTextViewController package before. This package is already updated.

___
Hi @rafaelks, I updated:

- [x] The 'blue bubble' appear to be smoother. This animation is slower when the right button is in screen bottom (without keyboard).
- [x] SlackTextViewController reference commit.

Thank you.

**Comment by rafaelks** [comment]
@falcaoaugustos Thank you! Looks like you're using the "wrong" branch of the SlackTextViewController. Can you update it? We updated it few days ago.

**Comment by rafaelks** [comment]
@falcaoaugustos The commit you're adding to the Podfile doesn't have support to the `right` button. If I download your branch, I can't see the audio button. Can you?

**Comment by falcaoaugustos** [comment]
@rafaelks You can't see the audio button because I sent the right button PR on `chore/swift4_xcode9_ios11` branch of `https://github.com/rafaelks/SlackTextViewController/` and now Rocket.Chat.iOS repo uses the `master` branch.
Do I have to send a new PR on the `master` branch?

**Comment by rafaelks** [comment]
@falcaoaugustos Great!!! Now I see it working, sorry about that. The feature is looking very good! I've a few suggestions to be made before we merge this (we really want to merge this :-)):

1. Try changing the UIButton target to just a `UITapGesture` and handling the event state (begin, end, etc) to update the CircleView; The way it is today on **device** it sometimes delays ~1.5 seconds to show the circle and start recording. Needs to be instantaneously to feel good;
2. If 1 works, remove the `delaysTouchesBegan` changes and see if it works;
3. Change `AudioServicesPlaySystemSound(kSystemSoundID_Vibrate)` to use the [UIFeedbackGenerator](https://developer.apple.com/documentation/uikit/uifeedbackgenerator) class. This way we use the Taptic from the iPhones instead of just vibrating. Feels a lot better IMHO;

Very very anxious to merge this one! 👍 🎉

**Comment by cardoso** [comment]
Hey! Just an update on this:

I found that the changes in the TextViewController pod were not needed and were introducing breaking changes to other branches.

So I reverted them and updated this branch to not need them :+1:

**Comment by falcaoaugustos** [comment]
There is a problem to implement UIFeedbackGenerator class when using microphone. You can see here about it: https://forums.developer.apple.com/thread/65301

We can do it later.

**Comment by rafaelks** [comment]
@falcaoaugustos Sounds good, let's leave `UIFeedbackGenerator` to a later version.

**Comment by rafaelks** [comment]
Closing this one from #1061

**Pull request closed without merging**